packages/server/lib/middleware/resource-capping.middleware.ts

```diff
@@ -0,0 +1,37 @@
+import type { Request, Response, NextFunction } from 'express';
+import {
+    accountService,
+    getEnvironmentAndAccountId,
+    errorManager,
+    connectionCreationStartCapCheck as connectionCreationStartCapCheckHook
+} from '@nangohq/shared';
+
+export const authCheck = async (req: Request, res: Response, next: NextFunction) => {
+    const { success, error, response } = await getEnvironmentAndAccountId(res, req);
+
+    if (!success || response === null) {
+        errorManager.errResFromNangoErr(res, error);
+        return;
+    }
+
+    const { accountId, environmentId } = response;
+
+    const account = await accountService.getAccountById(accountId);
```

Comment:
I like the idea in theory, but the issue is that the value of `is_capped` can change via the api and then the value stored in the `res.locals` is stale and potentially inaccurate. Since the user changing the `is_capped` setting is different than the user who has the capping nor not.

Reply:
Ah yes, but still this is making it more complicated than it needs since right now the only controller it goes through is the `connection.controller` when it is imported. If it went into the `access.middleware` instead of the `capping.middleware` then the logic to check the cap would need to go through
- `oauth.controller`
- `unauth.controller`
- `apiAuth.controller`
- `appStoreController`